### ISSUE TITLE
Version 9.4.0 is not compatible with Scala 3.2.2

### ISSUE THREAD

**Issue by seakayone** [jwt-scala/jwt-scala#486]
When using version 9.4.0 in a Scala 3.2.2 project I get the following compile errors:
```
[error] 46 |  private val alg      = Seq(JwtAlgorithm.HS256)
[error]    |                             ^^^^^^^^^^^^^^^^^^
[error]    |              value HS256 is not a member of object pdi.jwt.JwtAlgorithm
[error] error while loading JwtAlgorithm$,
[error] class file pdi/jwt/JwtAlgorithm.class is broken, reading aborted with class dotty.tools.tasty.UnpickleException
[error] TASTy signature has wrong version.
[error]  expected: {majorVersion: 28, minorVersion: 2}
[error]  found   : {majorVersion: 28, minorVersion: 3}
[error] 
[error] This TASTy file was produced by a more recent, forwards incompatible release.
[error] To read this TASTy file, please upgrade your tooling.
[error] The TASTy file was produced by Scala 3.3.0.
```

The exact same code compiles with Scala 3.3.0 which I currently cannot use for other reasons. 
I solved my issue by downgrading to 9.3.0 which works fine with the same code in Scala 3.2.2.

**Comment by erwan** [comment]
Yes, this is because other libs we depend on have already bumped to 3.3.0.

The solution is to bump your app to Scala 3.3.0, which can be done easily and without risk.

**Comment by seakayone** [comment]
I am using a combination of the sbt-native package and zio-json which fails on 3.3.0.
It is not that easy.

**Comment by seakayone** [comment]
Guess I should consider switching to a Java library instead rather than risk unsupported version incompatibilities in the future when sticking with jwt-scala.

**Comment by erwan** [comment]
Yes, I was surprised by Scala's choice to have no forward compatibility with Scala 3 versions too, but because we depend on so many json libraries we have no choice but to use the highest Scala version used by them.

You can stick with 9.3.0 until your problem on sbt-native and zio-json is fixed for 3.3.0, or as you say if a Java library works better for you then by all means use that.

**Comment by seakayone** [comment]
> no forward compatibility with Scala 3 versions

:D At least it is better now than with Scala 2. 

I appreciate your fast answers. We are using the jwt-scala lib in two of our projects and would like to stick with it.

Turns out [I could solve the problem I had](https://github.com/zio/zio-config/pull/1171#issuecomment-1600846611), it was a problem with `zio-config` not `zio-json`.

Apologies if I sounded a bit harsh, I did not mean to sound threatening. Coming from a Java background I still have to get used to the more complicated compiler versioning compatibility thing.
